app/src/main/java/com/example/todoapp/common/RestException.kt

```diff
@@ -0,0 +1,14 @@
+package com.example.todoapp.common
+
+import java.io.IOException
+
+/**
+ * Ошибки при запросах к серверу.
+ */
+sealed class RestException(val code: Int) : IOException() {
+    data object BadCredentials : RestException(400)
+    data object NotAuthorize : RestException(401)
+    data object NotFound : RestException(404)
+    data class InternalServerError(override val message: String?) : RestException(500)
+    data class UnexpectedRest(override val message: String?) : RestException(1000)
```

Comment:
Магические числа. Можно вынести в константы или 
```suggestion
    data object BadCredentials : RestException(code = 400)
    data object NotAuthorize : RestException(code = 401)
    data object NotFound : RestException(code = 404)
    data class InternalServerError(override val message: String?) : RestException(code = 500)
    data class UnexpectedRest(override val message: String?) : RestException(code = 1000)
```

И, вроде, эти коды даже не используются